### PR TITLE
Extension support and close event for V10

### PR DESCRIPTION
**Pull request by RangerMauve**
Added in some of the changes I did to the master branch so that they can be included in the v10 release. 😁

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -120,10 +120,27 @@ A boolean indicating whether the archive is writable.
...
+
+#### `archive.on('extension', name, message, peer)`
+
+Emitted when a peer sends you an extension message with `archive.extension()`.
+You can respond with `peer.extension(name, message)`.
+
+#### `archive.extension(name, message)`
+
+Send an extension message to connected peers. [Read more in the hypercore docs](https://github.com/mafintosh/hypercore#feedextensionname-message).
+
```
Comment (mafintosh) [inline README.md]: where is this implemented?

**Comment by RangerMauve** [comment]
Whoops. Good catch!

**Comment by mafintosh** [comment]
@RangerMauve Gah, it *did* auto close because I deleted the branch. What a weird default! Can you reopen against master? Sorry about the troubles

**Comment by mafintosh** [comment]
@RangerMauve Also I think you need to send the extension on all the metadata feeds that are mounted?

**Pull request closed without merging**